k9mail-library/src/test/java/com/fsck/k9/mail/store/pop3/Pop3FolderTest.java

```diff
@@ -0,0 +1,329 @@
+package com.fsck.k9.mail.store.pop3;
+
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.fsck.k9.mail.FetchProfile;
+import com.fsck.k9.mail.FetchProfile.Item;
+import com.fsck.k9.mail.Folder;
+import com.fsck.k9.mail.Folder.FolderType;
+import com.fsck.k9.mail.MessageRetrievalListener;
+import com.fsck.k9.mail.MessagingException;
+import com.fsck.k9.mail.internet.BinaryTempFileBody;
+import com.fsck.k9.mail.store.StoreConfig;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyZeroInteractions;
+import static org.mockito.Mockito.when;
+
+
+public class Pop3FolderTest {
+    private Pop3Store mockStore;
+    private Pop3Connection mockConnection;
+    private StoreConfig mockStoreConfig;
+    private MessageRetrievalListener<Pop3Message> mockListener;
+    private Pop3Folder folder;
+
+    @Before
+    public void before() throws MessagingException {
+        mockStore = mock(Pop3Store.class);
+        mockConnection = mock(Pop3Connection.class);
+        mockStoreConfig = mock(StoreConfig.class);
+        mockListener = mock(MessageRetrievalListener.class);
+        when(mockStore.getConfig()).thenReturn(mockStoreConfig);
+        when(mockStoreConfig.getInboxFolderName()).thenReturn("Inbox");
+        when(mockStore.createConnection()).thenReturn(mockConnection);
+        when(mockConnection.executeSimpleCommand(Pop3Commands.STAT_COMMAND)).thenReturn("+OK 10 0");
+        folder = new Pop3Folder(mockStore, "Inbox");
+        setupTempDirectory();
+    }
+
+    private void setupTempDirectory() {
```

Comment:
In another test class we use `BinaryTempFileBody.setTempDirectory(new File(System.getProperty("java.io.tmpdir")));`. That seems to work well enough.